plugins/producer/triedump.go

```diff
@@ -12,27 +12,20 @@ import (
 	gtypes "github.com/ethereum/go-ethereum/core/types"
 	"github.com/ethereum/go-ethereum/core/state"
 	"github.com/ethereum/go-ethereum/log"
-	// "github.com/ethereum/go-ethereum/rlp"
 	"github.com/ethereum/go-ethereum/rpc"
 	"github.com/ethereum/go-ethereum/trie"
-	// "github.com/ethereum/go-ethereum/common/hexutil"
-	// "github.com/ethereum/go-ethereum/crypto"
-
-	// "github.com/openrelayxyz/plugeth-utils/core"
-	// "github.com/openrelayxyz/plugeth-utils/restricted/types"
-	// "github.com/openrelayxyz/plugeth-utils/restricted/rlp"
 )
 
 var (
 	headerCache *lru.Cache
 )
 
 func getTrie(hash common.Hash) (state.Trie, error) {
-	stateDB, _, err := backend.StateAndHeaderByNumberOrHash(context.Background(), rpc.BlockNumberOrHashWithHash(hash, true))
+	tr, err := trie.NewStateTrie(trie.TrieID(common.Hash(hash)), trie.NewDatabase(backend.ChainDb(), nil))
```

Comment:
I don't think this actually works as well as the StateAndHeaderByNumberOrHash approach.

This getTrie can only get the trie if it was actually committed to disk, and all the data is found in `backend.ChainDB()`. In normal operation, Geth starts a single instance of `trie.NewDatabase(...)` and keeps a bunch of stuff in memory until it reaches a threshold and commits some of it to disk. The `StateAndHeaderByNumberOrHash()` approach gets it off of Geth's `trie.Database` instance, which has access to the in-memory stuff, while this approach only has access to what is on disk.

What you have here is the approach we originally took in Geth, then at some point we updated how we do it in Foundation Geth, but that never made its way into Bor. As it stands, this implementation only really works for one block every couple of hours, and the over-arching goal of the trie dump implementation is to compare successive blocks, which it ultimately can't really do with this implementation.

I get that the StateAndHeaderByNumberOrHash approach may not translate cleanly to Bor, but I think we need to figure out how to get access to the in memory Trie database, rather than taking this approach. (We might be able to use the singleton approach).